### DISCUSSION TITLE
Why Transitions only work once?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#10714] [Q&A] [answered]
I want to change the height with animation by checkbox. Here is my code:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication1.MainWindow"
        Title="AvaloniaApplication1">
	<Window.Styles>
		<Style Selector="Border.TestChecked">
			<Setter Property="Height" Value="150"/>
		</Style>
		<Style Selector="Border.TestUnChecked">
			<Setter Property="Height" Value="0"/>
		</Style>
	</Window.Styles>
	<Grid RowDefinitions="auto,auto">
		<CheckBox Name="CB"></CheckBox>
		<Border Width="100" HorizontalAlignment="Left" Background="Red" Classes.TestChecked="{Binding #CB.IsChecked}" Classes.TestUnChecked="{Binding !#CB.IsChecked}" Grid.Row="1">
			<Border.Transitions>
				<Transitions>
					<DoubleTransition Property="Height" Duration="0:0:0.5"></DoubleTransition>
				</Transitions>
			</Border.Transitions>
		</Border>
	</Grid>
</Window>
```

![1](https://user-images.githubusercontent.com/30720036/226242553-bfaebb57-717f-45c6-92a9-078763e3009a.gif)

However, as you see above. When I uncheck the checkbox, Transitions will not work. What's wrong with my code?

**Comment by timunie** [comment, marked as answer]
Seems like it doesn't work using Style-classes. Using a converter it works for me.

```xml
<Border Width="100" HorizontalAlignment="Left" Background="Red" 
        Height="{Binding #CB.IsChecked, Converter={x:Static vm:Conv.Current}}"
        Classes.TestChecked="{Binding #CB.IsChecked}" Classes.TestUnChecked="{Binding !#CB.IsChecked}" Grid.Row="1">
    <Border.Transitions>
        <Transitions>
            <DoubleTransition Property="Height" Duration="0:0:0.5" ></DoubleTransition>
        </Transitions>
    </Border.Transitions>
</Border>
```

```cs
public class Conv : IValueConverter
{
    public static Conv Current { get; } = new();
    
    public object? Convert(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        return value as bool? == true ? 100 : 0;
    }

    public object? ConvertBack(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        throw new NotImplementedException();
    }
}
```

**Reply by kyuranger** [reply]
> Seems like it doesn't work using Style-classes. Using a converter it works for me.
> 
> ```
> <Border Width="100" HorizontalAlignment="Left" Background="Red" 
>         Height="{Binding #CB.IsChecked, Converter={x:Static vm:Conv.Current}}"
>         Classes.TestChecked="{Binding #CB.IsChecked}" Classes.TestUnChecked="{Binding !#CB.IsChecked}" Grid.Row="1">
>     <Border.Transitions>
>         <Transitions>
>             <DoubleTransition Property="Height" Duration="0:0:0.5" ></DoubleTransition>
>         </Transitions>
>     </Border.Transitions>
> </Border>
> ```
> 
> ```cs
> public class Conv : IValueConverter
> {
>     public static Conv Current { get; } = new();
>     
>     public object? Convert(object? value, Type targetType, object? parameter, CultureInfo culture)
>     {
>         return value as bool? == true ? 100 : 0;
>     }
> 
>     public object? ConvertBack(object? value, Type targetType, object? parameter, CultureInfo culture)
>     {
>         throw new NotImplementedException();
>     }
> }
> ```

Achieving by converter it works now. However, as you mention in issue, it is a bug that cannot change height by style, hopes it can be fixed in the future. Thank you.
